### PR TITLE
MM-15670 Mark as Unread

### PR DESCRIPTION
**Pull request by hmhealey**
This is the PR to merge the mark-as-unread feature branch into master. It's made up of the following PRs:
- #3427 Adds the Mark as Unread option to the dot menu
- #3526 Reorganizes the dot menu as per the Mark as Unread spec
- #3512 Adds the ability to mark a channel as unread by alt+clicking on a post
- #3609 Adds a websocket event for multiple sessions to sync manually unread channels
- #4014 Stops marking a manually unread channel as read when the current user sends a message
- #4055 Stops marking a manually unread channel as read when the app regains focus
- #4149 Adds Mark as Unread options to the RHS
- #4184 Fixes a bug with the post menu in mobile web view
- #4138 Reverts some earlier behaviour to prevent a user from marking a channel as unread from their own posts (to be changed later)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15670

#### Related PRs
Mobile PR: https://github.com/mattermost/mattermost-mobile/pull/3554
Redux PR: https://github.com/mattermost/mattermost-redux/pull/986
Server PR: https://github.com/mattermost/mattermost-server/pull/13082

### CODE REVIEW

File: components/post_view/post/post.jsx [hunk 1]
```diff
@@ -234,7 +252,15 @@ class Post extends React.PureComponent {
     }
 
     unsetHover = () => {
-        this.setState({hover: false});
+        this.setState({hover: false, alt: false});
+    }
+
+    handleAlt = (e) => {
```
Comment (devinbinnie) [inline components/post_view/post/post.jsx]: duplicate function?

**Comment by hmhealey** [comment]
Blocked on https://github.com/mattermost/mattermost-webapp/pull/4213 as well

File: components/post_view/post/post.jsx (hunk 1 above)
Comment (hmhealey) [reply components/post_view/post/post.jsx]: Oh yeah, I noticed that at some point, but forgot to fix it. I'll fix it once that other PR is merged

**Comment by hmhealey** [comment]
That last commit also included the update to mattermost-redux

**Pull request merged**